File: android/app/src/main/java/com/kurt/bgmate/data/remote/BggApiRemoteDataSource.kt
package com.kurt.bgmate.data.remote

import com.kurt.bgmate.domain.model.BoardGame
import javax.inject.Inject

class BggApiRemoteDataSource @Inject constructor(
    private val bggApiService: BggApiService
) : BggRemoteDataSource {

    override suspend fun searchGames(query: String): List<BoardGame> {
        return try {
            val xmlString = bggApiService.searchGames(query)
            BggXmlParser.parseSearchResult(xmlString).map { it.toDomain() }
        } catch (e: Exception) {
            emptyList()
        }
    }

    override suspend fun fetchThumbnails(ids: List<String>): Map<String, String> {
        if (ids.isEmpty()) return emptyMap()
        return ids.chunked(20)
            .flatMap { chunk ->
                try {
                    val xmlString = bggApiService.getGameDetail(chunk.joinToString(","))
                    BggXmlParser.parseThingThumbnails(xmlString).entries
                } catch (e: Exception) {
                    emptyList()
                }
            }
            .associate { it.key to it.value }
    }
}

File: flutter/lib/data/remote/bgg_api_remote_data_source.dart
import 'package:bgmate_flutter/data/remote/bgg_api_service.dart';
import 'package:bgmate_flutter/data/remote/bgg_remote_data_source.dart';
import 'package:bgmate_flutter/domain/model/board_game.dart';

import 'bgg_xml_parser.dart';

class BggApiRemoteDataSource implements BggRemoteDataSource {
  final BggApiService _apiService;

  BggApiRemoteDataSource(this._apiService);

  @override
  Future<List<BoardGame>> searchGames(String query) async {
    final xml = await _apiService.searchGames(query);
    return BggXmlParser.parseSearchResults(xml);
  }
}
